tests/flex/test_pythonfunctioncontainer.py

```diff
@@ -61,3 +66,19 @@ def test_with_executor_wait(self):
             self.assertFalse(job.server.future.done())
             self.project.wait_for_job(job=job, interval_in_s=0.01, max_iterations=1000)
             self.assertTrue(job.server.future.done())
+
+    def test_with_internal_executor(self):
+        job = self.project.wrap_python_function(my_function_exe)
+        job.input["a_lst"] = [1, 2, 3, 4]
+        job.input["b_lst"] = [5, 6, 7, 8]
+        job.server.cores = 2
+        with self.assertRaises(TypeError):
+            job.executor_type = "Executor"
+        with self.assertRaises(TypeError):
+            job.executor_type = ProcessPoolExecutor
+        with self.assertRaises(TypeError):
+            job.executor_type = None
```

Comment:
Blocking change: seeing this test, it actually becomes clear to me that this is the opposite behaviour we want. `job._executor_type = None` is the default state, which is to not use an executor, so I believe we explicitly _do_ want to allow this value to be set in order to allow users to turn off executor functionality.

Reply:
I added the option to reset the executor.